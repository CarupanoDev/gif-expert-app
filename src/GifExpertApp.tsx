import { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"


const GifExpertApp = ({defaultCategories = []}: {defaultCategories: string[]}) => {

  const [categories, setCategories] = useState <string[]>(defaultCategories)

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />

      <ol>
        {categories.map(category => (
          <GifGrid
            key = {category}
            category={category}
          />
        ))}
      </ol>
    </>
  )
}

export default GifExpertApp
