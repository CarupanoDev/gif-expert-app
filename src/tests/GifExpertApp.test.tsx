import {describe, test, expect} from "vitest"
import {shallow} from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Test on <GifExpertApp/>', () =>{
  test('It should show the component correctly', () =>{
    const wrapper = shallow(<GifExpertApp defaultCategories={[]}/>)

    expect(wrapper).toMatchSnapshot()
  })

  test('It should show a list of categories', () =>{
    const categories = ['Inuyasha', 'Batman']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
