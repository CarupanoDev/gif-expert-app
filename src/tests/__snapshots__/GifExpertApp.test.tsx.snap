// Vitest Snapshot v1

exports[`Test on <GifExpertApp/> > It shoul show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on <GifExpertApp/> > It should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Inuyasha"
      key="Inuyasha"
    />
    <GifGrid
      category="Batman"
      key="Batman"
    />
  </ol>
</Fragment>
`;

exports[`Test on <GifExpertApp/> > It should show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
